lib/apimlStorage.ts

```diff
@@ -0,0 +1,382 @@
+/*
+  This program and the accompanying materials are
+  made available under the terms of the Eclipse Public License v2.0 which accompanies
+  this distribution, and is available at https://www.eclipse.org/legal/epl-v20.html
+  
+  SPDX-License-Identifier: EPL-2.0
+  
+  Copyright Contributors to the Zowe Project.
+*/
+
+import * as https from 'https';
+import * as http from 'http';
+import axios from 'axios';
+import { AxiosInstance } from 'axios';
+
+let apimlClient: AxiosInstance;
+
+export function configure(settings: ApimlStorageSettings) {
+  apimlClient = axios.create({
+    baseURL: `https://${settings.host}:${settings.port}`,
+    httpsAgent: new https.Agent(settings.tlsOptions)
+  });
+}
+
+export function isConfigured(): boolean {
+  return apimlClient != null;
+}
+
+export interface ApimlStorageSettings {
+  host: string;
+  port: number;
+  tlsOptions: https.AgentOptions;
+}
+
+
+export function makeStorageForPlugin(pluginId: string): ApimlStorage {
+  return new ApimlStorage(pluginId);
+}
+
+const CACHING_SERVICE_URI = '/cachingservice/api/v1/cache';
```

Comment:
everything on apiml that i have used follows the pattern
/api/v1
/ui/v1/
/ws/v1
Does caching service really start with `/cachingservice`? Thats interesting. Needs a code comment about that.

Reply:
Yes, it really does.